code/modules/antagonists/cult/cult_structures.dm

```diff
@@ -262,6 +272,250 @@
 			new N(get_turf(src))
 			to_chat(user, "<span class='cultitalic'>You summon the [choice] from the archives!</span>")
 
+/////////////////////////////WE'RE IN THE ENDGAME NOW/////////////////////////////
+/obj/structure/destructible/cult/pillar
+	name = "obsidian pillar"
+	icon_state = "pillar-enter"
+	icon = 'icons/obj/cult_64x64.dmi'
+	pixel_x = -16
+	obj_integrity = 200
+	max_integrity = 200
+	break_sound = 'sound/effects/meteorimpact.ogg'
+	break_message = "<span class='warning'>The pillar crumbles!</span>"
+	layer = MASSIVE_OBJ_LAYER
+	var/alt = 0
+	is_endgame = TRUE
+
+/obj/structure/destructible/cult/pillar/New()
+	..()
+	var/turf/T = loc
+	if (!T)
+		qdel(src)
+		return
+	for (var/obj/O in loc)
+		if(O == src)
+			continue
+		O.ex_act(2)
+		if(!QDELETED(O) && (istype(O, /obj/structure) || istype(O, /obj/machinery)))
+			qdel(O)
+	T.narsie_act()
+
+/obj/structure/destructible/cult/pillar/take_damage(damage_amount, damage_type = BRUTE, damage_flag = 0, sound_effect = 1, attack_dir, armour_penetration = 0)
+	..()
+	update_icon()
+
+/obj/structure/destructible/cult/pillar/Destroy()
+	new /obj/effect/decal/cleanable/ash(loc)
+	..()
+
+/obj/structure/destructible/cult/pillar/alt
+	icon_state = "pillaralt-enter"
+	alt = 1
+
+/obj/structure/destructible/cult/pillar/update_icon()
+	icon_state = "pillar[alt ? "alt": ""]2"
+	overlays.len = 0
```

Comment:
Use cut_overlays instead of setting len to 0 but... why do you need to clear overlays at all?